test/load/terraform/test-servers.tf

```diff
@@ -56,8 +56,9 @@ resource "aws_instance" "test-server" {
   subnet_id                   = (module.vpc.public_subnets)[0]
   provisioner "remote-exec" {
     inline = [
+      "sleep 2m",
```

Comment:
Why does this need to sleep?

Reply:
To allow time for Consul servers to stand up and everything to get connected.

Reply:
Is it possible to add a [health check](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html) on the target group instead? That way, this remote-exec won't run until `module.alb` is ready (when the health check passes). Something [like this](https://github.com/terraform-aws-modules/terraform-aws-alb/blob/master/examples/complete-alb/main.tf#L277-L287).

Reply:
The sleep will be brittle and depending on the sizing of instances/environment it might take longer.

Reply:
I think this should work:
```
  health_check {
    interval          = 5
    timeout           = 3
    protocol          = "HTTP"
    healthy_threshold = 2
    path              = "/v1/agent/self"
  }
  ```